### PR TITLE
Template Builder: Edit Box Content in Dialog

### LINKED ISSUE #25367: Template Builder: Edit Box Content in  Dialog
### User Story
As a CMS user, I want to be able to edit the content of a box when it is too small, so I can add and remove containers and classes.

### Acceptance Criteria
- The user should be able to open a dialog when the box is one-two columns.
- The user should be able to add and remove containers and classes in the dialog.

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
N/A

### Assumptions & Initiation Needs
- The dialog must be easy to use and understand.
- The content of the box should be updated in real time when changes are made.

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates:
N/A

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
* Allow the user to open a dialog when the box is one columns.

### Checklist
- [x] Tests
- [x] Translations

### <samp>🤖 Generated by Copilot at c1320af</samp>

Improved the template builder box component by adding a dialog for adding containers when the box width is small. Updated the dialog style and the box component logic and tests. Modified some scss and html files to use angular and primeng components and directives.

## Related Issue
Fixes #25367

# Explanation of Changes

### <samp>🤖 Generated by Copilot at c1320af</samp>

*  Refactor template builder box component to use ng-container and ng-templates for small and large box variants ([link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-bad3f14381a1639d24dc3958319e2ffaf13ad3342ea3276184e86b7fbd4c3e09L1-R82), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aL3-R3), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aL40-R45), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aL75-R87), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aR94-R97))
*  Replace p-dropdown component with p-button component for adding containers when box variant is small and trigger p-dialog component opening ([link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-bad3f14381a1639d24dc3958319e2ffaf13ad3342ea3276184e86b7fbd4c3e09L67-R94), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aL65-R70), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aL75-R87), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aR94-R97))
*  Add p-dialog component to render large box variant content when box variant is small and dialog is visible ([link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-bad3f14381a1639d24dc3958319e2ffaf13ad3342ea3276184e86b7fbd4c3e09L1-R82), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-769c98d8aaa736f86942f3d9d20765bb660348e4938b8911baf548f0572f94c7R9), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-769c98d8aaa736f86942f3d9d20765bb660348e4938b8911baf548f0572f94c7L35-R37), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aR16), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aL40-R45), [link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-522ebd82bd93c99880b22d8a3dddaf00afd05b8814a1d6c969f121493333014aL65-R70))
*  Modify style of dialog close button to match design specifications ([link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-123bc2019d92d71d5d2d93da360b73b30eefd6efb45c2c4686e92f6ed2995c68R85-R90))
*  Reduce font size of dialog title to match design specifications ([link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-123bc2019d92d71d5d2d93da360b73b30eefd6efb45c2c4686e92f6ed2995c68L22-R22))
*  Add test cases to verify dialog opening and closing behavior for small and large box variants ([link](https://github.com/dotCMS/core/pull/25393/files?diff=unified&w=0#diff-769c98d8aaa736f86942f3d9d20765bb660348e4938b8911baf548f0572f94c7R208-R260))

### Screenshots

https://github.com/dotCMS/core/assets/72418962/d3e23c87-d51a-4865-90cc-4d7e50f4e7ff

### CODE REVIEW

File: core-web/package-lock.json
Comment (zJaaal) [inline core-web/package-lock.json:1]: Do we need this?

File: core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.ts [hunk 1]
```diff
@@ -72,16 +77,23 @@ export class TemplateBuilderBoxComponent implements OnChanges {
...
     }
 
     onContainerSelect({ value }: { value: DotContainer }) {
         this.addContainer.emit(value);
         this.formControl.setValue(null);
     }
+
+    boxSizeMatches(size: TemplateBuilderBoxSize): boolean {
+        return this.boxVariant === this.templateBuilderSizes[size];
+    }
```
Comment (zJaaal) [inline core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.ts:98]: Are we going to use this?

**Review by zJaaal — COMMENTED** [review]
.

File: core-web/package-lock.json
Comment (rjvelazco) [reply core-web/package-lock.json:1]: MB, I'll. change it.

File: core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.ts:98]: I was testing something, I forgot to remove it.

**Review by manuelrojas — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by fmontes — COMMENTED** [review]
~On hold until we figure it out with Melissa.~

**Review by jdotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**